src/extension/prompts/node/agent/agentPrompt.tsx

```diff
@@ -126,19 +128,23 @@ export class AgentPrompt extends PromptElement<AgentPromptProps> {
 					endpoint={this.props.endpoint}
 					tools={this.props.promptContext.tools?.availableTools}
 					enableCacheBreakpoints={this.props.enableCacheBreakpoints}
+					userQueryTagName={userQueryTagName}
```

Comment:
It seems like there's a lot of plumbing of passing around all these individual things via props. It made it hard to find where they are actually defined and used. I'd suggest either just resolve them at the place where they are needed instead of early on, or pass it around as one unit, or part of promptContext, so we don't have to add the noise of many individual props.